src/validators/index.js

```diff
@@ -0,0 +1,42 @@
+const PropTypes = require('prop-types')
+
+//The guid validator checks the length of the prop since the guid format is consistent.
+//I can't think of a situation where we're trying to pass through a incorrect prop
+//that's a string of 36 characters.
+
+function chainedGuidLengthChecker(isRequired) {
+  return function(props, propName, componentName) {
+    const guid = props[propName] || ''
+
+    if (!guid && isRequired) {
```

Comment:
The [guid lib](https://npm.im/guid) that we are doing exposes a `.isGuid` method which would simplify this quite substantially.